airflow/api/common/experimental/mark_tasks.py

```diff
@@ -145,6 +145,9 @@ def set_state(task, execution_date, upstream=False, downstream=False,
                         dr.state = state
                         session.merge(dr)
 
+                if drs and commit:
```

Comment:
@bolkedebruin, if I don't do this I have a PrimaryKey conflict exception from sqlalchemy. If I commit here after merge - everything is fine.